src/select/select.js

```diff
@@ -107,82 +97,92 @@ class Select extends React.Component<PropsT, StatelessStateT> {
 
   onMouseLeave = (e: SyntheticEvent<HTMLInputElement>) => {};
 
+  onTextInputChange = (e: SyntheticEvent<HTMLInputElement>) => {
+    // $FlowFixMe
+    const newTextValue = e.target.value;
+    this.setState({
+      textValue: newTextValue,
+    });
+    this.props.onTextInputChange(e);
+    this.loadOptions(newTextValue).then(() => {
+      this.setState({isDropDownOpen: true});
+      if (this.props.filterable) {
+        let filteredOptions = this.state.options.filter(option =>
+          this.props.filterOption(option, newTextValue),
```

Comment:
`filterable` says whether or not we should filter options based on the text input value, and `filterOption` lets you override the function that determines whether an option matches the given query. You may want to set `filterable={true}` but not pass a `filterOption` (at which point we default to a simple lowercase string match). You may also want to set `filterable={false}` if you want to disable our internal filtering logic and somehow do it yourself outside the component.

Combining would probably mean `filterOption: boolean | (OptionT) => boolean` – in theory it would work but hard to say whether this actually simplifies the interface or just makes it more confusing.